Packs/PassiveTotal/Integrations/PassiveTotal_v2/PassiveTotal_v2.py

```diff
@@ -0,0 +1,871 @@
+    :param key: main key of sub-dictionary (e.g "foo")
+    :return: flat dictionary with pascal formatted keys (e.g. {"FooBar": "some-value"})
+    """
+
+    flat_dict: Dict[str, str] = {}
+    for sub_key, sub_value in src.items():
+        pascal_key = '{}{}'.format(key, sub_key[0].upper() + sub_key[1:])
+        flat_dict[pascal_key] = sub_value
+
+    return flat_dict
+
+
+def prepare_context_dict(response_dict: Dict[str, Any],
+                         keys_with_hierarchy: tuple = (),
+                         exclude_keys: tuple = ()) -> Dict[str, str]:
+    """
+    Prepare the context dictionary as per the standards.
+
+    :param response_dict: dictionary getting from API response that contains sub-dictionaries
+    :param keys_with_hierarchy: list of keys that contains sub-dictionary as its value.
+    :param exclude_keys: keys need to exclude.
+    :return: single level dictionary
+    """
+    simple_dict: Dict[str, str] = {}
+    for key, value in response_dict.items():
+        if key in keys_with_hierarchy:
+            simple_dict.update(nested_to_flat(response_dict.get(key, {}), key))
+        elif key not in exclude_keys:
+            simple_dict[key] = value
+    return simple_dict
+
+
+def get_context_for_whois_commands(domains: List[Dict[str, Any]]) -> Tuple[list, list]:
+    """
+    Prepare context for whois commands.
+
+    :param domains: list of domains return from response
+    :return: standard context and custom context for whois commands
+    """
+    standard_context_domain_list: List[Common.Domain] = []
+    custom_context: List[Dict[str, Any]] = []
+    for domain in domains:
+        # set domain standard context
+        standard_context_domain = Common.Domain(
+            domain=domain.get('domain', ''),
+            creation_date=domain.get('registered', ''),
+            updated_date=domain.get('registryUpdatedAt', ''),
+            expiration_date=domain.get('expiresAt', ''),
+            name_servers=domain.get('nameServers', ''),
+            organization=domain.get('organization', ''),
+            admin_name=domain.get('admin', {}).get('name', ''),
+            admin_email=domain.get('admin', {}).get('email', ''),
+            admin_phone=domain.get('admin', {}).get('telephone', ''),
+            admin_country=domain.get('admin', {}).get('country', ''),
+            registrar_name=domain.get('registrar', ''),
+            registrant_email=domain.get('registrant', {}).get('email', ''),
+            registrant_name=domain.get('registrant', {}).get('name', ''),
+            registrant_phone=domain.get('registrant', {}).get('telephone', ''),
+            registrant_country=domain.get('registrant', {}).get('country', ''),
+            dbot_score=Common.DBotScore(
+                indicator=domain.get('domain', ''),
+                indicator_type=DBotScoreType.DOMAIN,
+                integration_name=INTEGRATION_NAME,
+                score=Common.DBotScore.NONE
```

Comment:
Because it is mandatory to create it. Ref: https://github.com/demisto/content/blob/master/Packs/Base/Scripts/CommonServerPython/CommonServerPython.py#L2369